test/Constraints/diagnostics.swift

```diff
@@ -1415,6 +1415,107 @@ let f12: (Int) -> Void = { _ in }
 func f12<T : P2>(_ n: T, _ f: @escaping (T) -> T) {}
 f12(3, f4)// expected-error {{extra argument in call}}
 
+func testLabelErrorsBasic() {
```

Comment:
I wonder if since there are quite a few tests specifically about label matching, is it a good idea to separate them in their own label matching file? This is just an idea because I see this `diagnostics` file is general, and maybe separate label matching diagnostics can make it easier to organize things :)

Reply:
I think it's a good idea.

Test cases about matching of call arguments are also in `test/Constraints/keyword_arguments.swift`.

I don't know how other authors distinguish them.

I choice `diagnostics.swift` for what has interest in unlabeled arguments because they has no *keyword*.

If we can make a big change,
I want to rename `keyword_arguments.swift` to `argument_matching.swift` and 
move some cases in `diagnostics.swift` to it.

Reply:
If we can, I want to split it more like:

- `argument_matching_labeled.swift`
- `argument_matching_unlabeled.swift`
- `argument_matching_variadic.swift`
- `argument_matching_defaulted.swift`
- `argument_matching_trailing_closure.swift`

Reply:
I think since all fall in the context of `argument_matching` I think only one file is already fine.  
I don't know exactly what are the rules to this in the test suit, but to me, it seems reasonable to separate in, for example, an `argument_matching.swift` and have the subcases e.g. labeled, unlabeled, variadic.. divided inside the file.

Reply:
I think extracting all this into `test/Constraints/argument_matching.swift` would be just fine.

Reply:
@LucianoPAlmeida @xedin Thanks to comment.
OK I will make new patch to create `argument_matching.swift`.